### ISSUE TITLE
Feature request: lock some of the layers alternately in training

### ISSUE THREAD

**Issue by saintthor** [tensorflow/tensorflow#34675]
For example, a model with 3 dense layers named L0, L1, L2. Train it in below steps:

1, Set all layers trainable, train it for 10 epochs.
2, Set L0 not trainable, train it for 10 epochs.
3, Set L0 trainable, set L1 not trainable, train it for 10 epochs.
4, Set L1 trainable, set L2 not trainable, train it for 10 epochs.
5, Like step 2, set L2 trainable, set L0 not trainable, train it for 10 epochs.
6, Like step 3.
.......

It seems to train in this way is better than to train all the layers together. I think to reduce the trainable layers may reduce the gradient disappearance.

But for each time to alter the layers I have to recompile the whole model. I want there is some feature to do it automatically something like dropout.

If there already has this feature, please notice me.

**Comment by gowthamkpr** [comment]
@saintthor You can take layers from a model instance and freeze them. To "freeze" a layer means to exclude it from training, i.e. its weights will never be updated. And yes for this, you have to compile the model again.

As of now there is no feature that does what you are expecting to do.

**Comment by saintthor** [comment]
> @saintthor You can take layers from a model instance and freeze them. To "freeze" a layer means to exclude it from training, i.e. its weights will never be updated. And yes for this, you have to compile the model again.
> 
> As of now there is no feature that does what you are expecting to do.

To freeze is to set Layer.trainable = False?  I did it before. thanks.

**Comment by karmel** [comment]
Recompiling is the recommended way to do this, or you can look at creating custom layers/optimizers for this. However, it is too complicated a use-case to bake in to the keras lib at this time.
